packages/core/src/shared/utilities/workspaceUtils.ts

```diff
@@ -268,7 +268,7 @@ export function checkUnsavedChanges(): boolean {
     return vscode.workspace.textDocuments.some((doc) => doc.isDirty)
 }
 
-export function getExcludePattern(additionalPatterns: string[] = []) {
+export function getExcludePattern(additionalPatterns: string[] = [], nonGlobalExtraPatterns: boolean = true) {
```

Comment:
could be less convoluted.

The fact that `extraPatterns` looks redundant after `additionalPatterns` is just signaling problems with this interface design. Those problems aren't avoided by using convoluted parameter names.

```suggestion
export function getExcludePattern(additionalPatterns: string[] = [], extraPatterns: boolean = true) {
```

Reply:
good call, yeah, we shouldn't have too many params. I actually noticed there is no usage for `additionalPatterns`, so only the `defaultExcludePatterns` is sufficient. updated the naming to better reflect the behavior.